### PR TITLE
 Added shared library for Testcases. Useful for GUI, for example.

### PR DESCRIPTION
**Pull request by eriknellessen**
Testcases are then callable as functions.

### CODE REVIEW

File: eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.cpp [hunk 1]
```diff
@@ -1064,14 +1066,15 @@ int getAuthenticationParamsAutentApp(const char *const SP_URL,
 	size_t sz_len = sizeof sz;
 
 	connection_status = eIDClientConnectionStartHttp(&connection, SP_URL, NULL, NULL, DontGetHttpHeader, DontFollowHttpRedirect);
-
+
```
Comment (frankmorgner) [inline eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.cpp]: please remove trailing whitespaces

File: eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.cpp [hunk 2]
```diff
@@ -1110,6 +1113,47 @@ void printStatistics(int retValue, unsigned int size, int serverErrorCounter){
...
+	if(retValue != NPACLIENT_ERROR_SUCCESS) return retValue;
+	if(g_samlResponseReturncode != 0) return NPACLIENT_ERRO;
+
+#if SAML_VERSION == SAML_2 || SAML_VERSION == NO_SAML || defined SELBSTAUSKUNFT_WUERZBURG || defined AUTENTAPP
+	retValue = getSamlResponse2(response);
+#endif
+	return retValue;
+}
+
+#ifndef AS_LIBRARY
```
Comment (frankmorgner) [inline eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.cpp]: Could you please create separate files if possible? there are already way to much ifdef's...

File: eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.h [hunk 3]
```diff
@@ -0,0 +1,7 @@
+int performEID(std::string strServiceURL,
```
Comment (frankmorgner) [inline eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.h]: when creating a new file you should also review the visual studio project files

File: eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.cpp (hunk 1 above)
Comment (eriknellessen) [inline eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.cpp]: Done.

File: eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.cpp (hunk 2 above)
Comment (eriknellessen) [inline eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.cpp]: I think, that this is an issue independent of the pull request. I suggest to merge the pull request and to solve this issue in  #64

File: eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.h (hunk 3 above)
Comment (eriknellessen) [inline eIDClientCore/bin/Test_nPAClientLib/Test_nPAClientLib.h]: I think, that this is an issue independent of the pull request. I suggest to merge the pull request and to solve this issue in  #65

**Comment by frankmorgner** [comment]
agreed

**Comment by frankmorgner** [comment]
could you resolve the merge conflicts?

**Pull request merged**